File: Data Structures & Algorithms/majority-element/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        d={}
        avg=len(nums)/2
        for i in nums:
            if i in d:
                d[i]+=1
            else:
                d[i]=1
        for x in d:
            if d[x]>avg:
                return x
        

File: Data Structures & Algorithms/maximum-product-subarray/submission-0.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int maxsofar=nums[0];
        int curmax=nums[0];
        int curmin=nums[0];
        for(int i=1;i<nums.size();i++){
            int num=nums[i];
            if(num<0){
                swap(curmax,curmin);
            }
            curmax=max(num,num*curmax);
            curmin=min(num,num*curmin);
            maxsofar=max(maxsofar,curmax);

        }
        return maxsofar;
    }
};
